compiler/luci-interpreter/src/kernels/Prelu.cpp

```diff
@@ -128,25 +149,97 @@ void Prelu::evalQuantized() const
+                                                             identity_mult.shift)
+                     : tflite::MultiplyByQuantizedMultiplier(
+                           input_val * alpha_val, alpha_mult.multiplier, alpha_mult.shift);
+  const int32_t clamped_output = std::min(quantized_max, std::max(quantized_min, output_val));
+  return clamped_output;
+}
+
+void Prelu::evalQuantizedS16() const
+{
+  bool channel_wise = alpha()->scales().size() != 1;
+  if (!channel_wise)
+  {
+    auto fn = [this](int16_t input_val, int16_t alpha_val) {
+      const ChannelQuantMultipliers pos_mult{_output_shift_identity, _output_multiplier_identity};
+      const ChannelQuantMultipliers &neg_mult = _alpha_multipliers[0];
+      return evalElemS16Prelu(input_val, alpha_val, pos_mult, neg_mult);
+    };
+
+    BinaryOpBroadcastSlow(getTensorShape(input()), getTensorData<int16_t>(input()),
+                          getTensorShape(alpha()), getTensorData<int16_t>(alpha()),
+                          getTensorShape(output()), getTensorData<int16_t>(output()), fn);
+  }
+  else
+  {
+    tflite::RuntimeShape input_shape = getTensorShape(input());
+    tflite::RuntimeShape alpha_shape = getTensorShape(alpha());
+    tflite::RuntimeShape output_shape = getTensorShape(output());
+    const int16_t *input_data = input()->data<int16_t>();
+    const int16_t *alpha_data = alpha()->data<int16_t>();
+    int16_t *output_data = output()->data<int16_t>();
+
+    constexpr int N = 5;
+
+    const ChannelQuantMultipliers pos_mult{_output_shift_identity, _output_multiplier_identity};
+
+    if (input_shape == alpha_shape)
+    {
+      const int flat_size = tflite::MatchingElementsSize(input_shape, alpha_shape, output_shape);
+      int quantized_dim = alpha()->quantized_dimension();
+
+      int32_t outer_dims_size = 1;
+      for (int i = 0; i < quantized_dim; ++i)
+        outer_dims_size *= input_shape.Dims(i);
+      int32_t quant_dim_size = alpha_shape.Dims(alpha()->quantized_dimension());
+      int32_t inner_dims_size = flat_size / outer_dims_size / quant_dim_size;
+
+      for (int32_t outer_dims = 0; outer_dims < outer_dims_size; ++outer_dims)
+        for (int32_t quant_channel = 0; quant_channel < quant_dim_size; ++quant_channel)
+        {
+          const ChannelQuantMultipliers &neg_mult = _alpha_multipliers[quant_channel];
+          for (int32_t inner_dims = 0; inner_dims < inner_dims_size; ++inner_dims)
+          {
+            size_t offset =
+                inner_dims + (quant_channel + outer_dims * quant_dim_size) * inner_dims_size;
+            output_data[offset] =
+                evalElemS16Prelu(input_data[offset], alpha_data[offset], pos_mult, neg_mult);
+          }
+        }
+    }
+    else
+    {
+      assert(input_shape.DimensionsCount() <= N);
+      assert(alpha_shape.DimensionsCount() <= N);
+      assert(output_shape.DimensionsCount() <= N);
```

Comment:
Maybe N = 4 ? (currently N=5)

Reply:
I took this limitation somewhere from tflite (but not from prelu, probably it was some binary operation, like sub).
Current implementation in tflite 2.3 sets N to 4, so you are right.